### PR TITLE
[tests-only][full-ci] tests: add api tests for deleting public link password

### PR DESCRIPTION
**Pull request by SwikritiT**
part of: https://github.com/owncloud/ocis/issues/7823

### CODE REVIEW

File: tests/acceptance/features/apiSpacesShares/shareOperations.feature [hunk 1]
```diff
@@ -351,3 +351,40 @@ Feature: sharing
...
+
+
+  Scenario Outline: remove password of a public link with change permission as a Space Admin
+    Given using spaces DAV path
+    And the administrator has assigned the role "Space Admin" to user "Alice" using the Graph API
+    And user "Alice" has created folder "FOLDER"
+    And user "Alice" has created a public link share with settings
+      | path        | /FOLDER  |
+      | permissions | <permissions>   |
+      | password    | %public% |
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiSpacesShares/shareOperations.feature:363]:
```suggestion
      | path        | /FOLDER       |
      | permissions | <permissions> |
      | password    | %public%      |
```

File: tests/acceptance/features/apiSpacesShares/shareOperations.feature [hunk 2]
```diff
@@ -351,3 +351,40 @@ Feature: sharing
...
+    When user "Alice" updates the last public link share using the sharing API with
+      | path        | /FOLDER |
+      | permissions | <permissions>  |
+      | password    |         |
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiSpacesShares/shareOperations.feature:367]:
```suggestion
      | path        | /FOLDER       |
      | permissions | <permissions> |
      | password    |               |
```

File: tests/acceptance/features/apiSpacesShares/shareOperations.feature [hunk 3]
```diff
@@ -351,3 +351,40 @@ Feature: sharing
...
+
+
+  Scenario: remove password of a public link with read permission as a Space Admin
+    Given using spaces DAV path
+    And the administrator has assigned the role "Space Admin" to user "Alice" using the Graph API
+    And user "Alice" has created folder "FOLDER"
+    And user "Alice" has created a public link share with settings
+      | path        | /FOLDER  |
+      | permissions | read   |
+      | password    | %public% |
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiSpacesShares/shareOperations.feature:384]:
```suggestion
      | path        | /FOLDER  |
      | permissions | read     |
      | password    | %public% |
```

File: tests/acceptance/features/apiSpacesShares/shareOperations.feature [hunk 4]
```diff
@@ -351,3 +351,40 @@ Feature: sharing
...
+    When user "Alice" updates the last public link share using the sharing API with
+      | path        | /FOLDER |
+      | permissions | read  |
+      | password    |         |
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiSpacesShares/shareOperations.feature:388]:
```suggestion
      | path        | /FOLDER |
      | permissions | read    |
      | password    |         |
```

File: tests/acceptance/features/apiSpacesShares/shareSpacesViaLink.feature [hunk 5]
```diff
@@ -128,3 +128,65 @@ Feature: Share spaces via link
...
+      | permissions | 1        |
+      | password    | %public% |
+    When user "Alice" updates the last public link share using the sharing API with
+      | permissions | 1 |
+      | password    |   |
+    Then the HTTP status code should be "200"
+    And the OCS status code should be "100"
+
+
+  Scenario Outline: remove password of a public link space share with read permission as a Space Admin
```
Comment (PrajwolAmatya) [inline tests/acceptance/features/apiSpacesShares/shareSpacesViaLink.feature:143]: Is it read permission for this scenario and for this scenario in L175?

Comment (SwikritiT) [reply tests/acceptance/features/apiSpacesShares/shareSpacesViaLink.feature:143]: No, it's not. I copied and pasted and forgot to update I think. Thanks, I'll update them!

File: tests/acceptance/expected-failures-API-on-OCIS-storage.md [hunk 6]
```diff
@@ -515,5 +515,8 @@ Not everything needs to be implemented for ocis. While the oc10 testsuite covers
 - [coreApiShareCreateSpecialToShares2/createShareDefaultFolderForReceivedShares.feature:22](https://github.com/owncloud/ocis/blob/master/tests/acceptance/features/coreApiShareCreateSpecialToShares2/createShareDefaultFolderForReceivedShares.feature#L22)
 - [coreApiShareCreateSpecialToShares2/createShareDefaultFolderForReceivedShares.feature:23](https://github.com/owncloud/ocis/blob/master/tests/acceptance/features/coreApiShareCreateSpecialToShares2/createShareDefaultFolderForReceivedShares.feature#L23)
 
+#### [Normal users can update the public link to delete its password if permission is not sent in data](https://github.com/owncloud/ocis/issues/7821)
+- [coreApiSharePublicLink1/changingPublicLinkShare.feature:150](https://github.com/owncloud/ocis/blob/master/tests/acceptance/features/coreApiSharePublicLink1/changingPublicLinkShare.feature#L150)
```
Comment (saw-jan) [inline tests/acceptance/expected-failures-API-on-OCIS-storage.md:519]:
```suggestion
#### [Normal users can update the public link to delete its password if permission is not sent in data](https://github.com/owncloud/ocis/issues/7821)

- [coreApiSharePublicLink1/changingPublicLinkShare.feature:150](https://github.com/owncloud/ocis/blob/master/tests/acceptance/features/coreApiSharePublicLink1/changingPublicLinkShare.feature#L150)
```

File: tests/acceptance/features/apiSpacesShares/shareOperations.feature [hunk 7]
```diff
@@ -351,3 +351,40 @@ Feature: sharing
       | permissions |
       | change      |
       | create      |
+
+
+  Scenario Outline: remove password of a public link with change permission as a Space Admin
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesShares/shareOperations.feature:356]:
```suggestion
  Scenario Outline: try to remove password of a public link with change permission as a Space Admin
```

File: tests/acceptance/features/apiSpacesShares/shareOperations.feature [hunk 8]
```diff
@@ -351,3 +351,40 @@ Feature: sharing
...
+    Then the HTTP status code should be "200"
+    And the OCS status code should be "400"
+    And the OCS status message should be "missing required password"
+    Examples:
+      | permissions |
+      | change      |
+      | create      |
+
+
+  Scenario: remove password of a public link with read permission as a Space Admin
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesShares/shareOperations.feature:377]: how about this way?
```suggestion
  Scenario: remove password of a public link share (read permission) as a Space Admin
```

File: tests/acceptance/features/apiSpacesShares/shareOperations.feature (hunk 7 above)
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesShares/shareOperations.feature:356]:
```suggestion
  Scenario Outline: remove password of a public link share (change/create permission) as a Space Admin
```

File: tests/acceptance/features/apiSpacesShares/shareSpacesViaLink.feature [hunk 9]
```diff
@@ -128,3 +128,65 @@ Feature: Share spaces via link
...
+    And the OCS status code should be "400"
+    And the OCS status message should be "missing required password"
+    Examples:
+      | permissions |
+      | 5           |
+      | 15          |
+      | 4           |
+
+
+  Scenario: remove password of a public link space share with read permission as a Space Admin
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesShares/shareSpacesViaLink.feature:160]: the scenario title has duplication
```suggestion
  Scenario: remove password of a public link space share with read permission as a Space Manager
```

File: tests/acceptance/features/apiSpacesShares/shareSpacesViaLink.feature [hunk 10]
```diff
@@ -128,3 +128,65 @@ Feature: Share spaces via link
...
+      | password    | %public% |
+    When user "Brian" updates the last public link share using the sharing API with
+      | permissions | 1 |
+      | password    |   |
+    Then the HTTP status code should be "200"
+    And the OCS status code should be "104"
+    And the OCS status message should be "user is not allowed to delete the password from the public link"
+
+
+  Scenario Outline: remove password of a public link space share with various permissions as a Space Admin
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesShares/shareSpacesViaLink.feature:175]:
```suggestion
  Scenario Outline: remove password of a public link space share with various permissions as a Space Manager
```

File: tests/acceptance/features/apiSpacesShares/shareOperations.feature [hunk 11]
```diff
@@ -351,3 +351,40 @@ Feature: sharing
       | permissions |
       | change      |
       | create      |
+
+
+  Scenario Outline: try to remove password of a public link share change/create permission) as a Space Admin
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesShares/shareOperations.feature:356]:
```suggestion
  Scenario Outline: try to remove password of a public link share (change/create permission) as a Space Admin
```

File: tests/acceptance/features/apiSpacesShares/shareSpacesViaLink.feature [hunk 12]
```diff
@@ -128,3 +128,66 @@ Feature: Share spaces via link
...
+      | permissions | 1        |
+      | password    | %public% |
+    When user "Alice" updates the last public link share using the sharing API with
+      | permissions | 1 |
+      | password    |   |
+    Then the HTTP status code should be "200"
+    And the OCS status code should be "100"
+
+
+  Scenario Outline: remove password of a public link space share with various permissions as a Space Admin
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesShares/shareSpacesViaLink.feature:144]:
```suggestion
  Scenario Outline: try to remove password of a public link space share with various permissions as a Space Admin
```

File: tests/acceptance/features/apiSpacesShares/shareSpacesViaLink.feature [hunk 13]
```diff
@@ -128,3 +128,66 @@ Feature: Share spaces via link
...
+      | password    | %public% |
+    When user "Brian" updates the last public link share using the sharing API with
+      | permissions | 1 |
+      | password    |   |
+    Then the HTTP status code should be "200"
+    And the OCS status code should be "104"
+    And the OCS status message should be "user is not allowed to delete the password from the public link"
+
+
+  Scenario Outline: remove password of a public link space share with various permissions as a Space Manager
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesShares/shareSpacesViaLink.feature:176]:
```suggestion
  Scenario Outline: try to remove password of a public link space share with various permissions as a Space Manager
```

File: tests/acceptance/features/coreApiSharePublicLink1/changingPublicLinkShare.feature [hunk 14]
```diff
@@ -131,3 +131,69 @@ Feature: changing a public link share
     When the public deletes file "parent.txt" from the last public link share using the password "%public%" and new public WebDAV API
     Then the HTTP status code should be "403"
     And as "Alice" file "PARENT/parent.txt" should exist
+
+
+  Scenario Outline:try to remove password of a public link share(change, create permission) as a normal user
```
Comment (saw-jan) [inline tests/acceptance/features/coreApiSharePublicLink1/changingPublicLinkShare.feature:136]: we used to write `<actor> does something`. maybe this way the titles can be clearer.
```suggestion
  Scenario Outline: normal user tries to remove password of a public link share (change, create permission)
```

File: tests/acceptance/features/apiSpacesShares/shareSpacesViaLink.feature [hunk 15]
```diff
@@ -128,3 +128,66 @@ Feature: Share spaces via link
...
+      | role      | manager |
+    And user "Brian" has created a public link share of the space "share space" with settings:
+      | permissions | 1        |
+      | password    | %public% |
+    When user "Brian" updates the last public link share using the sharing API with
+      | permissions | 1 |
+      | password    |   |
+    Then the HTTP status code should be "200"
+    And the OCS status code should be "104"
+    And the OCS status message should be "user is not allowed to delete the password from the public link"
```
Comment (SwikritiT) [inline tests/acceptance/features/apiSpacesShares/shareSpacesViaLink.feature:173]: This is a peculiar case where the error message and OCS status code is different. If I'm not wrong `1xx` is the success for `ocs` status code? The error message on this one is however very nice. I have written this scenario as the server  behaves right now as I don't know if this is an expected response or not.
Should the error code be `400` like in other similar cases what about the error message? 
cc @ScharfViktor @saw-jan

File: tests/acceptance/features/apiSpacesShares/shareOperations.feature [hunk 16]
```diff
@@ -351,3 +351,40 @@ Feature: sharing
...
+    Then the HTTP status code should be "200"
+    And the OCS status code should be "400"
+    And the OCS status message should be "missing required password"
+    Examples:
+      | permissions |
+      | change      |
+      | create      |
+
+
+  Scenario: space admin removes password of a public link with share (read permission)
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesShares/shareOperations.feature:377]:
```suggestion
  Scenario: space admin removes password of a public link share (read permission)
```

File: tests/acceptance/features/apiSpacesShares/shareSpacesViaLink.feature [hunk 17]
```diff
@@ -128,3 +128,66 @@ Feature: Share spaces via link
     Then the HTTP status code should be "400"
     And the OCS status message should be "Can not share space root"
     And for user "Alice" the space "Alice Hansen" should not contain the last created public link
+
+
+  Scenario: space admin removes password of a public link share (read permission) of a space
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesShares/shareSpacesViaLink.feature:133]:
```suggestion
  Scenario: space admin removes password of a public link share of space (read permission) 
```

File: tests/acceptance/features/apiSpacesShares/shareSpacesViaLink.feature [hunk 18]
```diff
@@ -128,3 +128,66 @@ Feature: Share spaces via link
...
+      | permissions | 1        |
+      | password    | %public% |
+    When user "Alice" updates the last public link share using the sharing API with
+      | permissions | 1 |
+      | password    |   |
+    Then the HTTP status code should be "200"
+    And the OCS status code should be "100"
+
+
+  Scenario Outline: space admin tries to remove password of a public link space share with various permissions
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesShares/shareSpacesViaLink.feature:144]:
```suggestion
  Scenario Outline: space admin tries to remove password of a public link share of space (various permissions)
```

File: tests/acceptance/features/apiSpacesShares/shareSpacesViaLink.feature [hunk 19]
```diff
@@ -128,3 +128,66 @@ Feature: Share spaces via link
...
+    And the OCS status code should be "400"
+    And the OCS status message should be "missing required password"
+    Examples:
+      | permissions |
+      | 5           |
+      | 15          |
+      | 4           |
+
+
+  Scenario: space manager tries to remove password of a public link space share (read permission)
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesShares/shareSpacesViaLink.feature:161]:
```suggestion
  Scenario: space manager tries to remove password of a public link share of space (read permission)
```

File: tests/acceptance/features/apiSpacesShares/shareSpacesViaLink.feature [hunk 20]
```diff
@@ -128,3 +128,66 @@ Feature: Share spaces via link
...
+      | password    | %public% |
+    When user "Brian" updates the last public link share using the sharing API with
+      | permissions | 1 |
+      | password    |   |
+    Then the HTTP status code should be "200"
+    And the OCS status code should be "104"
+    And the OCS status message should be "user is not allowed to delete the password from the public link"
+
+
+  Scenario Outline: space manager tries to remove password of a public link space share with various permissions
```
Comment (saw-jan) [inline tests/acceptance/features/apiSpacesShares/shareSpacesViaLink.feature:176]:
```suggestion
  Scenario Outline: space manager tries to remove password of a public link share of space (various permissions)
```

File: tests/acceptance/features/coreApiSharePublicLink1/changingPublicLinkShare.feature [hunk 21]
```diff
@@ -131,3 +131,69 @@ Feature: changing a public link share
     When the public deletes file "parent.txt" from the last public link share using the password "%public%" and new public WebDAV API
     Then the HTTP status code should be "403"
     And as "Alice" file "PARENT/parent.txt" should exist
+
+
+  Scenario Outline: normal user tries to remove password of a public link share (change, create permission)
```
Comment (saw-jan) [inline tests/acceptance/features/coreApiSharePublicLink1/changingPublicLinkShare.feature:136]:
```suggestion
  Scenario Outline: normal user tries to remove password of a public link share (change/create permission)
```

File: tests/acceptance/features/coreApiSharePublicLink1/changingPublicLinkShare.feature [hunk 22]
```diff
@@ -131,3 +131,69 @@ Feature: changing a public link share
...
+    When user "admin" updates the last public link share using the sharing API with
+      | path        | /PARENT |
+      | permissions | read    |
+      | password    |         |
+    Then the HTTP status code should be "200"
+    And the OCS status code should be "100"
+    And the public should be able to download file "/parent.txt" from inside the last public link shared folder using the new public WebDAV API with password ""
+
+
+  Scenario Outline: administrator tries to remove password of a public link share (change, create permission)
```
Comment (saw-jan) [inline tests/acceptance/features/coreApiSharePublicLink1/changingPublicLinkShare.feature:183]:
```suggestion
  Scenario Outline: administrator tries to remove password of a public link share (change/create permission)
```

**Review by saw-jan — COMMENTED** [review]
Generally looks good.

File: tests/acceptance/features/apiSpacesShares/shareSpacesViaLink.feature (hunk 15 above)
Comment (ScharfViktor) [reply tests/acceptance/features/apiSpacesShares/shareSpacesViaLink.feature:173]: Let's leave it at that. The main thing is that Brian can't delete the password. The web hides the password delete button for Brian, so the user can only try to delete the password using the api (he can see the message there).

**Review by ScharfViktor — APPROVED** [review]
LGTM

**Pull request merged**